tests/test_volume.py

```diff
@@ -26,450 +24,522 @@
-        elif self.dtype == np.float32:
-            new_dtype = np.float64
+
+def test_scalar_r_sub(vols_1, data_1):
+    result = 42 - vols_1
+    assert np.all(result == 42 - data_1)
+    assert isinstance(result, Volume)
+
+
+def test_scalar_mul(vols_1, data_2):
+    result = vols_1 * 123
+    assert np.all(result == data_2)
+    assert isinstance(result, Volume)
+
+
+def test_scalar_r_mul(vols_1, data_2):
+    result = 123 * vols_1
+    assert np.all(result == data_2)
+    assert isinstance(result, Volume)
+
+
+def test_scalar_div(vols_2, vols_1):
+    result = vols_2 / 123
+    assert np.allclose(result, vols_1)
+
+
+def test_right_scalar_div(vols_2, data_1):
+    result = 123 / vols_2
+    assert np.allclose(result, 1 / data_1)
+
+
+def test_div(vols_2, vols_1):
+    result = vols_2 / vols_1
+    assert np.allclose(result, 123)
+
+
+def test_right_div(data_2, vols_1):
+    result = data_2 / vols_1
+    np.allclose(result, 123)
+
+
+def test_save_load(vols_1):
+    # Create a tmpdir in a context. It will be cleaned up on exit.
+    with tempfile.TemporaryDirectory() as tmpdir:
+        # Save the Volume object into an MRC files
+        mrcs_filepath = os.path.join(tmpdir, "test.mrc")
+        vols_1.save(mrcs_filepath)
+
+        # Load saved MRC file as a Volume of dtypes single and double.
+        vols_loaded_single = Volume.load(mrcs_filepath, dtype=np.float32)
+        vols_loaded_double = Volume.load(mrcs_filepath, dtype=np.float64)
+
+        # Check that loaded data are Volume instances and compare to original volume.
+        assert isinstance(vols_loaded_single, Volume)
+        assert np.allclose(vols_1, vols_loaded_single)
+        assert isinstance(vols_loaded_double, Volume)
+        assert np.allclose(vols_1, vols_loaded_double)
+
+
+def test_project(vols_1, dtype):
+    if vols_1.shape[1] == 43:
+        pytest.xfail("Need to fix odd res projections being off.")
+    # first test with synthetic data
+    # Create a stack of rotations to test.
+    r_stack = np.empty((12, 3, 3), dtype=dtype)
+    for r, ax in enumerate(["x", "y", "z"]):
+        r_stack[r] = Rotation.about_axis(ax, 0).matrices
+        # We'll consider the multiples of pi/2.
+        r_stack[r + 3] = Rotation.about_axis(ax, pi / 2).matrices
+        r_stack[r + 6] = Rotation.about_axis(ax, pi).matrices
+        r_stack[r + 9] = Rotation.about_axis(ax, 3 * pi / 2).matrices
+
+    # Project a Volume with all the test rotations
+    vol_id = 1  # select a volume from Volume stack
+    img_stack = vols_1[vol_id].project(r_stack)
+
+    for r in range(len(r_stack)):
+        # Get result of test projection at center of Image.
+        prj_along_axis = img_stack.asnumpy()[r][21, 21]
+
+        # For Volume, take mean along the axis of rotation.
+        vol_along_axis = np.mean(vols_1.asnumpy()[vol_id], axis=r % 3)
+        # Volume is uncentered, take the mean of a 2x2 window.
+        vol_along_axis = np.mean(vol_along_axis[20:22, 20:22])
```

Comment:
Setting the resolution to 43 means we have a centered volume. On the next line we can do something like:
```
if vols_1.resolution % 2 == 1:
    vol_along_axis = vol_along_axis[vols_1.resolution // 2, vols_1.resolution // 2]
```
Then the test should pass for odd resolution. It would probably be good to change all the hardcoded values above (ie. 21, 20:22) to be in terms of `vols_1.resolution` instead. This way if we ever want to change `RES` we don't have to track down these hardcoded values.